src/UglyToad.PdfPig/Parser/FileStructure/FileHeaderOffset.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace UglyToad.PdfPig.Parser.FileStructure;
+
+/// <summary>
+/// How many bytes precede the "%PDF-" version header in the file. In some files this 'junk' can
+/// offset all following offset bytes.
+/// </summary>
+internal readonly struct FileHeaderOffset(int value)
```

Comment:
I think this would be a good use for `readonly **record** struct`, especially since you override the `Equals()` method. In order to avoid boxing, implementing `IEquatable<FileHeaderOffset>` might be necessary

Reply:
See https://nietras.com/2021/06/14/csharp-10-record-struct/ for me details and benchmark

Reply:
I don't think we can use this yet due to backwards compatibility, I get an error about init only properties being unsupported on-build.

Reply:
@Numpsy thanks for that, I believe we alredy use these polyfill in another parts of the lib. I'll take care of that in another PR